### ISSUE TITLE
Compare to Flow runtime?

### ISSUE THREAD

**Issue by svr93** [Jack-Works/proposal-enum#23]
Flow has different methods helpful for enum usage: https://flow.org/en/docs/enums/using-enums/#toc-methods
1) cast
2) isValid

It will be wonderful to include some helpers with strict type safety in TS:
```
v = SomeEnum.cast(raw) // v type inference: SomeEnum | null
```

**Comment by Jack-Works** [comment]
Yes, it's a useful feature but I'm worried about how ADT enums will work with those mehods.
